app/models/piece.rb

```diff
@@ -5,6 +5,19 @@ class Piece < ActiveRecord::Base
   validates :color, inclusion: { in: %w(black white) }
   validates :type, inclusion: { in: %w(Pawn Rook Bishop Knight King Queen) }
 
+  # Returns "valid" and updates the piece's x and
+  # y position to provided coordinates if move is valid,
+  # otherwise do nothing and return nil.
+  def move(x,y)
+    if valid_move?(x,y)
+      self.x_position = x
+      self.y_position = y
+      return "valid"
```

Comment:
I wasn't sure on the bang, so clearly I need to get a finer grained appreciation for that distinction.  I'lll make the executive decision to go back to having it return true/false, but keep it as a bang operator.  I think having the huh operator makes the method unclear as to its primary purpose and is why I went down this road of crapola 'valid'/nil in the first place as a misguided attempt to find a middle ground that was functional and stylistic.  I failed. :)